packages/vats/src/orchestration.js

```diff
@@ -0,0 +1,268 @@
+          powers.init(/** @type {keyof OrchestrationPowers} */ (name), power);
+        }
+      }
+      return { powers, icaControllerNonce: 0 };
+    },
+    {
+      admin: {
+        placeholder() {
+          return 'test';
+        },
+      },
+      public: {
+        /**
+         * @param {import('@agoric/orchestration').ConnectionId} hostConnectionId
+         *   the counterparty connection_id
+         * @param {import('@agoric/orchestration').ConnectionId} controllerConnectionId
+         *   self connection_id
+         * @param {Port} [currentPort] if a port is provided, it will be reused
+         *   to create the account
+         * @returns {Promise<ChainAccount>}
+         */
+        async createAccount(
+          hostConnectionId,
+          controllerConnectionId,
+          currentPort,
+        ) {
+          await null;
+
+          let port;
+          if (!matches(currentPort, M.remotable('Port'))) {
+            if (currentPort) trace('Invalid Port provided, binding a new one.');
+            const network = getPower(this.state.powers, 'network');
+            port = await E(network)
+              .bind(`/ibc-port/icacontroller-${this.state.icaControllerNonce}`)
+              .catch(e => Fail`Failed to get remoteAddress: ${e}`);
+
+            this.state.icaControllerNonce += 1;
+          } else {
+            port = currentPort;
+          }
+          const remoteConnAddr = makeICAConnectionAddress(
+            hostConnectionId,
+            controllerConnectionId,
+          );
+          const chainAccount = createChainAccount(port, remoteConnAddr);
+
+          // await so we do not return a ChainAccount before it successfully instantiates
+          await E(port)
+            .connect(remoteConnAddr, chainAccount.connectionHandler)
+            // XXX if we fail, should we close the port (if it was created in this flow)?
+            .catch(e => Fail`Failed to create ICA connection: ${bare(e)}`);
+
+          return chainAccount.account;
+        },
+        /** @param {string} _chainName e.g. cosmos */
+        getChain(_chainName) {
+          Fail`getChain not implemented yet`;
+        },
+      },
+    },
+  );
+
+/** @param {import('@agoric/base-zone').Zone} zone */
+export const prepareOrchestrationTools = zone => {
+  const createChainAccount = prepareChainAccount(zone);
+  const makeOrchestration = prepareOrchestration(zone, createChainAccount);
+
+  return harden({ makeOrchestration });
+};
+harden(prepareOrchestrationTools);
+
+/** @typedef {ReturnType<ReturnType<typeof prepareChainAccount>>} ChainAccountKit */
+/** @typedef {ChainAccountKit['account']} ChainAccount */
+/** @typedef {ReturnType<typeof prepareOrchestrationTools>} OrchestrationTools */
+/** @typedef {ReturnType<OrchestrationTools['makeOrchestration']>} OrchestrationKit */
+/** @typedef {OrchestrationKit['admin']} OrchestrationAdmin */
+/** @typedef {OrchestrationKit['public']} Orchestration */
```

Comment:
this name too general to bind to the public facet, imo. sites that need it could easily pull it off OrchestrationKit. I suppose the same for OrchestrationAdmin

Reply:
I removed `admin` and `OrchestrationAdmin` in [2a39e6d](https://github.com/Agoric/agoric-sdk/pull/9114/commits/2a39e6dc915c90b2a05be38311fdbcac418b6408).

I am wary of only exported `OrchestrationKit` - consumers of the API will likely only use the public facet, so the goal was to export a type that can be used to represent the `orchestration` available in the bootstrap space for a core proposal (e.g. stakeAtom). Would you prefer to see it called `OrchestrationPF`?